core/src/main/java/tc/oc/pgm/score/ScoreMatchModule.java

```diff
@@ -92,26 +95,60 @@ public double getScore(Competitor competitor) {
   }
 
   /** Gets the score message for the match. */
-  public String getScoreMessage() {
-    List<String> scores = Lists.newArrayList();
-    for (Entry<Competitor, Double> scorePair : this.scores.entrySet()) {
-      scores.add(scorePair.getKey().getColor().toString() + ((int) (double) scorePair.getValue()));
+  public Component getScoreMessage(MatchPlayer matchPlayer) {
+    List<Component> scoreMessages = Lists.newArrayList();
+    final FreeForAllMatchModule ffamm = match.getModule(FreeForAllMatchModule.class);
+    if (ffamm != null) {
+      scoreMessages =
+          this.scores.entrySet().stream()
+              .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+              .limit(10)
+              .map(
+                  x ->
+                      TextComponent.builder()
+                          .append(x.getKey().getName(NameStyle.VERBOSE))
+                          .append(TextComponent.of(": ", TextColor.GRAY))
+                          .append(TextComponent.of((int) x.getValue().doubleValue()))
+                          .color(TextColor.WHITE)
+                          .build())
+              .collect(Collectors.toList());
+    } else {
+
+      for (Entry<Competitor, Double> scorePair : this.scores.entrySet()) {
+        scoreMessages.add(
+            TextComponent.of(
+                ((int) scorePair.getValue().doubleValue()),
+                TextFormatter.convert(scorePair.getKey().getColor())));
+      }
     }
```

Comment:
You can simplify this using a reducer instead of a collector:
```java
    TextComponent result =
        this.scores.entrySet().stream()
            .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
            .limit(10)
            .reduce(
                TextComponent.builder(),
                (builder, entry) ->
                    builder
                        .append(entry.getKey().getName(NameStyle.VERBOSE))
                        .append(": ", TextColor.GRAY)
                        .append("" + entry.getValue().intValue(), TextColor.WHITE),
                ComponentBuilder::append)
            .build();
```

Reply:
Well, I need the output to be a list, so I can use `TextFormatter.list(...)` to list the names in a nice format: `Player1, Player2, ... , and PlayerN`. Using a collector seems like the proper way right?

Reply:
A joining collector may be a better alternative then, look at how the `Collectors.joining` collector works for strings and do a similar one for this

Reply:
Using Collectors.joining, there is no easy way to have the last element separated like `<element n-1>, and <element n>`, rather than just a comma. I don't understand what is wrong with the method I am currently using.

Reply:
@mrcookieunderscore13 Maybe you are looking for something like what is in ```TextFormatter#list``` ?

Reply:
@KingOfSquares I do use that method. I use the stream to make an ordered list of the players, and then a few lines down use `TextFormatter#list` to format it properly.

Reply:
Right, then I think I misunderstood the argument between you and Pablo